consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/SpuriousBehaviourTest.java

```diff
@@ -0,0 +1,166 @@
+import static tech.pegasys.pantheon.consensus.ibft.support.MessageReceptionHelpers.assertPeersReceivedNoMessages;
+import static tech.pegasys.pantheon.consensus.ibft.support.TestHelpers.createSignedCommitPayload;
+
+import tech.pegasys.pantheon.consensus.ibft.ConsensusRoundIdentifier;
+import tech.pegasys.pantheon.consensus.ibft.messagedata.IbftV2;
+import tech.pegasys.pantheon.consensus.ibft.payload.CommitPayload;
+import tech.pegasys.pantheon.consensus.ibft.payload.MessageFactory;
+import tech.pegasys.pantheon.consensus.ibft.payload.PreparePayload;
+import tech.pegasys.pantheon.consensus.ibft.payload.SignedData;
+import tech.pegasys.pantheon.consensus.ibft.support.NodeParams;
+import tech.pegasys.pantheon.consensus.ibft.support.RoundSpecificNodeRoles;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContext;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContextBuilder;
+import tech.pegasys.pantheon.consensus.ibft.support.ValidatorPeer;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.crypto.SECP256K1.KeyPair;
+import tech.pegasys.pantheon.crypto.SECP256K1.Signature;
+import tech.pegasys.pantheon.ethereum.core.Block;
+import tech.pegasys.pantheon.ethereum.core.Hash;
+import tech.pegasys.pantheon.ethereum.core.Util;
+import tech.pegasys.pantheon.ethereum.p2p.api.MessageData;
+import tech.pegasys.pantheon.ethereum.p2p.wire.RawMessage;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class SpuriousBehaviourTest {
+
+  final long blockTimeStamp = 100;
+  private final Clock fixedClock =
+      Clock.fixed(Instant.ofEpochSecond(blockTimeStamp), ZoneId.systemDefault());
+
+  // Test is configured such that a remote peer is responsible for proposing a block
+  private final int NETWORK_SIZE = 5;
+
+  // Configuration ensures remote peer will provide proposal for first block
+  private final TestContext context =
+      new TestContextBuilder()
+          .validatorCount(NETWORK_SIZE)
+          .indexOfFirstLocallyProposedBlock(0)
+          .clock(fixedClock)
+          .build();
+  private final ConsensusRoundIdentifier roundId = new ConsensusRoundIdentifier(1, 0);
+  private final RoundSpecificNodeRoles roles = context.getRoundSpecificRoles(roundId);
+
+  private Block proposedBlock = context.createBlockForProposalFromChainHead(0, 30);
+  private SignedData<PreparePayload> expectedPrepare;
+  private SignedData<CommitPayload> expectedCommit;
+
+  @Before
+  public void setup() {
+    context.getController().start();
+
+    expectedPrepare =
+        context
+            .getLocalNodeMessageFactory()
+            .createSignedPreparePayload(roundId, proposedBlock.getHash());
+    expectedCommit =
+        createSignedCommitPayload(
+            roundId, proposedBlock, context.getLocalNodeParams().getNodeKeyPair());
+  }
+
```

Comment:
Do these two illegal RLP cover the control flow if junk RLP (RLP that doesn't 'fit' into the expected data structure)?

Reply:
The first test ensures RLP which doesn't match a given message ID doesn't prevent ongoing IBFT operation.

The latter test ensures if the seal in a commit msg is bad, that it is ignored.